node/src/types/block.rs

```diff
@@ -759,6 +761,20 @@ impl BlockHeader {
 
     /// Hash of the block header.
     pub fn hash(&self) -> BlockHash {
+        if self.protocol_version >= HASH_V2_PROTOCOL_VERSION {
+            self.hash_v2()
+        } else {
+            self.hash_v1()
+        }
+    }
+
+    fn hash_v1(&self) -> BlockHash {
+        let serialized_header = Self::serialize(&self)
+            .unwrap_or_else(|error| panic!("should serialize block header: {}", error));
```

Comment:
Isn't that equivalent to:
```suggestion
            .expect("should serialize block header");
```
Or does that use `Debug` instead of `Display`? (Also below.)

Reply:
I've checked, that uses `Debug`.